src/slice/new_iter.rs

```diff
@@ -0,0 +1,495 @@
+//! Okay, this immutable iterator doesn't use any `Access` types, and thus could not be used
+//! on any element with possible mutable access at the same time (or could, but it's a subtle UB).
+//! It could be used without UB only with unaliased memory, and due to this, `.iter()` should produce
+//! a group of `(AliasedIter, UnAliasedIter, AliasedIter)` or use AliasedIter for every `.next()`.
+
+// This should be deleted when all unsafe blocks are used or deleted.
+#![allow(unused_unsafe)]
+
+#[cfg(feature = "std")]
+use std::{
+	iter::{
+		Chain, DoubleEndedIterator, ExactSizeIterator, FusedIterator, Iterator,
+	},
+	marker::PhantomData,
+	mem,
+	ptr::NonNull,
+};
+
+#[cfg(not(feature = "std"))]
+use core::{
+	iter::{
+		Chain, DoubleEndedIterator, ExactSizeIterator, FusedIterator, Iterator,
+	},
+	marker::PhantomData,
+	mem,
+	ptr::NonNull,
+};
+
+use crate::{
+	access::BitAccess,
+	index::{BitIdx, BitSel},
+	mem::BitMemory,
+	order::BitOrder,
+	pointer::BitPtr,
+	slice::BitSlice,
+	store::BitStore,
+};
+
+macro_rules! unchecked_add {
+	($x:expr, $y:expr) => {{
+		$x + $y
+		}};
+}
+
+macro_rules! unchecked_sub {
```

Comment:
Do these expressions actually remove checks? I'm going to switch to `wrapping_arith`, since we can rely on preconditions of the pointer representation to know that wraps will never actually occur.

Reply:
No, lol. But if there will be real use of `intrinsics::unchecked_sub()` only one line in macro should be changed.

Reply:
`.wrapping_op` methods produce their corresponding instruction without any other checks, and silently discard any overflow flags; I can't think of any particular reason to use any other intrinsic if we don't need to care about boundary conditions.

Reply:
> Returns the result of an unchecked subtraction, resulting in undefined behavior when `x - y > T::max_value()` or `x - y < T::min_value()`.

Not wrapping operation. Not like it's worth using anyway. Welp, it's pretty easy to change the name of macro.

Btw, still need help with copying aliased elements and getting `idx.select()` from `raw_parts()` index, which doesn't work right now due to some strange type thing.